addons/mail/models/ir_cron.py

```diff
@@ -0,0 +1,37 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, SUPERUSER_ID
+
+
+class IrCron(models.AbstractModel):
+    _inherit = 'ir.cron'
+
+    def _get_admin_channel(self):
+        """ Get the admin channel.
+        If it doesn't exist, creates it. """
+        admin_group = self.env.ref("base.group_system", raise_if_not_found=False)
+        if not admin_group:
+            return
+
+        domain = [
+            ("channel_type", "=", "channel"),
+            ("group_public_id", "=", admin_group.id),
+            ("group_ids", "in", admin_group.id),
+        ]
+
+        admin_channel = self.env['discuss.channel'].search(domain).sorted(lambda c: "admin" not in c.name.lower())
```

Comment:
Would say: bad idea :D better create a channel with an xml id so that we have a real way of finding it.

Reply:
:+1: The goal was to reuse already existing channel if possible (this code is similar to upgrade admin notification)

Reply:
FWIW, I agree with Thibault; the entire method could simply become `ref("mail.admin_channel")`, making it much simpler. Certainly, one could argue that this channel might be deleted, but this risk also applies to other records, such as groups or the Odoo bot. I believe we generally agree that master data should not be deleted.

Reply:
> 👍 The goal was to reuse already existing channel if possible (this code is similar to upgrade admin notification)

This will be the job of the upgrade script to assign the xmlid to such group.